### PR TITLE
[AT-2982] update guzzle

### PR DESCRIPTION
**Pull request by Cbelbegra**
Jira: https://careerbuilder.atlassian.net/browse/AT-2982

### CODE REVIEW

File: src/CareerBuilder/OAuth2/Flows/Flow.php [hunk 1]
```diff
@@ -45,15 +58,27 @@ protected function __construct(array $configs, ClientInterface $client = null, L
...
+        $stack = HandlerStack::create();
+        $stack->push(
+            Middleware::log(
+                $logger,
+                new MessageFormatter('{req_body} - {res_body}')
+            ),
+            'DMSApiClientLogger'
+        );
+
+        $stack =
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/Flow.php]: :x:

File: src/CareerBuilder/OAuth2/Flows/Flow.php [hunk 2]
```diff
@@ -118,10 +144,13 @@ public function getToken(AccessToken $token = null)
...
         }
 
-        $request = $this->client->post($this->tokenRequestPath, $this->headers, $this->body);
-        $response = $request->send();
-        $data = $response->json();
-
+        /** @var Response $response */
+        $response = $this->client->post($this->tokenRequestPath,[
+            'headers' => $this->headers,
+            'form_params' =>$this->body
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/Flow.php]: ❌Hey, this code is not following the PSR-12 standard, which we are trying to strongly follow in this codebase.
Doc: https://www.php-fig.org/psr/psr-12

File: src/CareerBuilder/OAuth2/Flows/Flow.php [hunk 3]
```diff
@@ -118,10 +144,13 @@ public function getToken(AccessToken $token = null)
             $this->buildBody();
         }
 
-        $request = $this->client->post($this->tokenRequestPath, $this->headers, $this->body);
-        $response = $request->send();
-        $data = $response->json();
-
+        /** @var Response $response */
+        $response = $this->client->post($this->tokenRequestPath,[
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/Flow.php]: Is this G6 compatible??

File: src/CareerBuilder/OAuth2/Flows/Flow.php [hunk 4]
```diff
@@ -45,15 +58,27 @@ protected function __construct(array $configs, ClientInterface $client = null, L
...
-        $this->client = $client ?: new Client();
-        $this->client->setBaseUrl($configs['base_url']);
-        $this->client->addSubscriber(new LogPlugin(new PsrLogAdapter($this->logger)));
+        $stack = HandlerStack::create();
+        $stack->push(
+            Middleware::log(
+                $logger,
+                new MessageFormatter('{req_body} - {res_body}')
+            ),
+            'DMSApiClientLogger'
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/Flow.php]: What's that? 😄

Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/Flow.php]: Create a client factory like other projects for this...

File: composer.json [hunk 5]
```diff
@@ -14,5 +16,8 @@
         "psr-0": {
             "CareerBuilder\\": "src/"
         }
-    }
+    },
+    "repositories": [
+        { "type": "vcs",    "url": "git@github.com:cbdr/json-manager.git" }
```
Comment (eexit) [inline composer.json]: Don't inject a new dependency... Just override PHP's function and add error throwing like stated [here](https://github.com/cbdr/candidate-search-api/pull/38#discussion_r365334634).

**Review by eexit — CHANGES_REQUESTED** [review]
No big deal but still require fixes.

File: src/CareerBuilder/OAuth2/Flows/Flow.php (hunk 3 above)
Comment (Cbelbegra) [reply src/CareerBuilder/OAuth2/Flows/Flow.php]: Yeah, tokenRequestPath is a string, and the parameters was named correctly

**Review by lassalle — APPROVED** [review]

File: src/CareerBuilder/OAuth2/Flows/Flow.php [hunk 6]
```diff
@@ -136,4 +145,35 @@ protected function getJWT($claims)
     {
         return JWT::encode($claims, $this->sharedSecret, 'HS512');
     }
+
+    /**
+     * Throws an exception upon JSON decode for PHP < 7.3
+     * @deprecated Remove when upgrading to PHP 7.3
+     */
+    function json_decode ($json, $assoc = false, $depth = 512, $options = 0) {
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/Flow.php]: 👉 This needs to be moved outside of the class.

File: src/CareerBuilder/OAuth2/Flows/FlowClientFactory.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+namespace CareerBuilder\OAuth2\Flows;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\MessageFormatter;
+use GuzzleHttp\Middleware;
+
+class FlowClientFactory
+{
+    public static function create($configs, $logger)
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/FlowClientFactory.php]: ☝️ Missing or invalid [phpDoc](https://docs.phpdoc.org/references/phpdoc/index.html).

File: src/CareerBuilder/OAuth2/Flows/FlowClientFactory.php [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            ),
+            'OAuthClientLogger'
+        );
+
+        return new Client([
+            'base_uri' => $configs['base_url'],
+            'handler' => $stack,
+        ]);
+    }
+}
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/FlowClientFactory.php]: ❌Hey, this code is not following the PSR-12 standard, which we are trying to strongly follow in this codebase.
Doc: https://www.php-fig.org/psr/psr-12

**Review by eexit — COMMENTED** [review]
Close enough, last pass and you'll be good 👍

File: src/CareerBuilder/OAuth2/Flows/FlowClientFactory.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+use GuzzleHttp\Middleware;
+
+class FlowClientFactory
+{
+    /**
+     * @param $configs
+     * @param $logger
+     * @return Client
+     */
+    public static function create($configs, $logger)
```
Comment (eexit) [inline src/CareerBuilder/OAuth2/Flows/FlowClientFactory.php]: No type hinting for the variables?

**Review by eexit — CHANGES_REQUESTED** [review]
There are still pending comments.

**Review by eexit — APPROVED** [review]
🚀All good!
Don't forget to fill in your PR description.

**Comment by eexit** [comment]
Closing in favor of https://github.com/careerbuilder/php-oauth/pull/13.

**Pull request closed without merging**